src/Command/ImportTerritorialCouncilFromCSVCommand.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace App\Command;
+
+use App\Entity\TerritorialCouncil\TerritorialCouncil;
+use App\Repository\ReferentTagRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Filesystem\Exception\FileNotFoundException;
+
+class ImportTerritorialCouncilFromCSVCommand extends Command
+{
+    protected static $defaultName = 'app:territorial-council:import-from-csv';
+
+    /** @var EntityManagerInterface */
+    private $em;
+    /** @var ReferentTagRepository */
+    private $referentTagRepository;
+    /** @var SymfonyStyle */
+    private $io;
+
+    public function __construct(EntityManagerInterface $em, ReferentTagRepository $referentTagRepository)
+    {
+        $this->em = $em;
+        $this->referentTagRepository = $referentTagRepository;
+
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->addArgument('file', InputArgument::REQUIRED, 'CSV file with territorial councils')
+        ;
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->io = new SymfonyStyle($input, $output);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->io->text('Start importing territorial councils');
+
+        $territorialCouncils = $this->parseCSV($input->getArgument('file'));
+
+        $this->io->progressStart(\count($territorialCouncils));
+
+        $i = 0;
+        $batchSize = 10;
+        foreach ($territorialCouncils as $data) {
+            ++$i;
+            if (null == $data['name']) {
+                $this->io->warning('Territorial council without name cannot be added.');
+
+                continue;
+            }
+
+            /** @var TerritorialCouncil $territorialCouncil */
+            if ($territorialCouncil = $this->em->getRepository(TerritorialCouncil::class)->findOneBy(['name' => $data['name']])) {
+                $territorialCouncil->clearReferentTags();
+            } else {
+                $territorialCouncil = new TerritorialCouncil($data['name'], $data['codes']);
+            }
+
+            $tags = explode(',', $data['tag_code']);
+            foreach ($tags as $tag) {
+                $tag = trim($tag);
+                if (!($referentTag = $this->referentTagRepository->findOneByCode($tag))) {
```

Comment:
maybe you can find all with codes : `findBy(['code' => explode(',', $data['tag_code'])` and browse results, so you have one query instead of one per code ?

Reply:
Yes, I can find all codes like this, but in this case I don't know if all passed referent tags have been found.
If I do one by one, I can display errors if there are in CSV file.